static/js/Autocomplete.jsx

```diff
@@ -0,0 +1,377 @@
+import Sefaria from "./sefaria/sefaria";
+import React, {useEffect, useState} from "react";
+import classNames from "classnames";
+import {SearchButton} from "./Misc";
+import { useCombobox } from 'downshift';
+
+const _type_icon_map = {
+  "Collection": "collection.svg",
+  "AuthorTopic": "iconmonstr-pen-17.svg",
+  "TocCategory": "iconmonstr-view-6.svg",
+  "PersonTopic": "iconmonstr-hashtag-1.svg",
+  "Topic": "iconmonstr-hashtag-1.svg",
+  "ref": "iconmonstr-book-15.svg",
+  "search": "iconmonstr-magnifier-2.svg",
+  "Term": "iconmonstr-script-2.svg",
+  "User": "iconmonstr-user-2%20%281%29.svg"
+};
+
+const _type_icon = function(itemType, itemPic) {
+    if (itemType === "User" && itemPic !== "") {
+      return itemPic;
+    } else {
+      return `/static/icons/${_type_icon_map[itemType]}`;
+    }
+};
+
+
+function groupByType(seggestedItems) {
+    const groupedItems = {};
+
+    // Group items by their "type", "Topic" and "PersonTopic" considered same type
+    seggestedItems.forEach(item => {
+        let itemType = item.type in ["Topic", "PersonTopic"] ? "Topic" : item.type;
+        if (!groupedItems[itemType]) {
+            groupedItems[itemType] = [];
+        }
+        groupedItems[itemType].push(item);
+    });
+
+    //Convert into a datastructure like this: [{"type": name,
+    //                                         "items" : [item1, item2]}]
+    const result = Object.keys(groupedItems).map(type => ({
+        type,
+        items: groupedItems[type]
+    }));
+    return result;
+};
+
+const getURLForObject = function(type, key) {
+    if (type === "Collection") {
+      return `/collections/${key}`;
+    } else if (type === "TocCategory") {
+      return `/texts/${key.join('/')}`;
+    } else if (type in {"Topic": 1, "PersonTopic": 1, "AuthorTopic": 1}) {
+      return `/topics/${key}`;
+    } else if (type === "ref") {
+      return `/${key.replace(/ /g, '_')}`;
+    } else if (type === "User") {
+      return `/profile/${key}`;
+    }
+};
+
+const getQueryObj = (query) => {
+  return Sefaria.getName(query)
+    .then(d => {
+      const repairedCaseVariant = Sefaria.repairCaseVariant(query, d);
+      if (repairedCaseVariant !== query) {
+        return getQueryObj(repairedCaseVariant);
+      }
+      const repairedQuery = Sefaria.repairGershayimVariant(query, d);
+      if (repairedQuery !== query) {
+        return getQueryObj(repairedQuery);
+      }
+
+      if (d["is_ref"]) {
+        return {'type': 'Ref', 'id': d["ref"]};
+      } else if (!!d["topic_slug"]) {
+        return {'type': 'Topic', 'id': d["topic_slug"]};
+      } else if (d["type"] === "Person" || d["type"] === "Collection" || d["type"] === "TocCategory") {
+        return {'type': d["type"], 'id': d["key"]};
+      } else {
+        return {'type': "Search", 'id': query};
+      }
+    });
+};
+
+
+const SearchSuggestion = ({ itemKey, itemType, itemLabel, itemUrl, itemPic }) => {
```

Comment:
Remove `item` from the prefix of these props. It should be understood that they refer to an item based on the context.

Reply:
`itemKey` seems to be unused